forest/parse_args.py

```diff
@@ -30,3 +30,11 @@ def add_arguments(parser):
     parser.add_argument(
         "--file-type", default="unified_model", metavar="FILETYPE",
         help="keyword to navigate/display file(s)")
+    parser.add_argument(
+        "--var", action="append", dest="variables", type=key_value,
+        help="variable to substitute in --config-file")
+
+
+def key_value(item):
+    head, *tail = item.split(":")
+    return head, ":".join(tail)
```

Comment:
We can get more rigorous syntax checking AND simplify the code by using the `nargs` attribute - that's if you're happy to accept a slight change in command line syntax:
```python
parser.add_argument(
    '--var', action='append', dest='variables', nargs=2, metavar=('KEY', 'VALUE'),
    help='...blah blah...')
```
The command line syntax would then be:
```bash
$ forest --var directory /foo/bar --var wibble wobble
```
And the syntax help comes out like:
```
usage: forest [-h] [--var KEY VALUE]

optional arguments:
  -h, --help       show this help message and exit
  --var KEY VALUE  blah blah...
```

Reply:
Ah, I had forgotten that `argparse` supports fixed `nargs`, I mainly use `+` and `*`. Great idea!

Reply:
That also supports Unix shell expansion, nice idea